packages/netlify-cms-core/src/components/Editor/EditorControlPane/Widget.js

```diff
@@ -10,7 +10,8 @@ const isEmpty = value =>
   value === null ||
   value === undefined ||
   (value.hasOwnProperty('length') && value.length === 0) ||
-  (value.constructor === Object && Object.keys(value).length === 0);
+  (value.constructor === Object && Object.keys(value).length === 0) ||
+  (List.isList(value) && value.size === 0);
```

Comment:
This will affect the current `list` widget as well. Before you could have a list that is required but it wouldn't mark it as invalid if no items were added.

Reply:
Awesome, good catch.